test-static-stdlib/test-dispatch-static-stdlib.test

```diff
@@ -0,0 +1,7 @@
+REQUIRES: platform=Linux
+RUN: rm -rf %t
+RUN: mkdir -p %t
+RUN: %{swiftc} -static-stdlib %S/dispatch_test.swift -lDispatchStubs -lbsd -o %t/dispatch_test
```

Comment:
`-lDispatchStubs` being necessary seems very suspect to me. Shouldn't that be handled by Dispatch itself?

Reply:
> @kevints is doing the release management for this release

This is targeted at `master`, so this should be reviewed by code owners instead.

Reply:
For that matter `-lbsd` seems suspect. Is the problem a missing modulemap for libbsd?

Reply:
`libdispatch.a` is an archive library so wont auto link `libswiftDispatch.a` and `libDispatchStubs.a` hence being explicitly listed. 

`-lbsd` was needed the last time I tested this to provide `strlcpy` and `getprogname` however it looks like they are now included as shims in the dispatch source tree so I have removed `-lbsd` from the test.

@kevints @jrose-apple Could someone please re-CI test  this PR, thanks

Reply:
Yeah, @kevints reminded me about what we can and can't do with static libraries. I just wasn't thinking.